packages/ses/src/error/assert.js

```diff
@@ -54,6 +54,50 @@ freeze(quote);
  */
 const hiddenDetailsMap = new WeakMap();
 
+/**
+ * @param {HiddenDetails} hiddenDetails
+ * @returns {string}
+ */
+const getMessageString = ({ template, args }) => {
+  const parts = [template[0]];
+  for (let i = 0; i < args.length; i += 1) {
+    const arg = args[i];
+    let argStr;
+    if (declassifiers.has(arg)) {
+      argStr = `${arg}`;
+    } else if (arg instanceof Error) {
+      argStr = `(${an(arg.name)})`;
+    } else {
+      argStr = `(${an(typeof arg)})`;
+    }
+    parts.push(argStr, template[i + 1]);
+  }
+  return parts.join('');
+};
+
+/**
+ * Give detailsTokens a toString behavior. To minimize the overhead of
+ * creating new detailsTokens, which is crucial, we do this with an
```

Comment:
I haven't seen any evidence that minimizing this overhead is crucial. It's intuitive that minimizing the overhead is worthwhile, so I'm not arguing against the optimization. But without evidence / measurements, the "which is crucial" bit seems superfluous. If we do have evidence, even anecdotal, perhaps add a few words about it.

Reply:
@FUDCo IIRC, the evidence was your's. A long long time ago --- before Covid. Do you remember?

Reply:
@dckc in any case, I removed the questionable text. Whether Chip remembers or not, the text was not needed. Resolving.

Reply:
I have a dim memory of such an issue, but my recollection (fuzzy!) is that it was one `details` in particular that was the source of the difficulty -- something that was generating the complicated details object essentially every pass through some inner loop.  I will attempt to reconstruct when I have a chance.